cmd/contour/serve.go

```diff
@@ -578,7 +582,7 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 		log := log.WithField("context", "xds")
 
 		log.Printf("waiting for informer caches to sync")
-		if !clients.WaitForCacheSync(stop) {
+		if !clients.WaitForCacheSync(serveContext) {
```

Comment:
Switch to `g.AddContext(...)` for this task too so the workgroup can manage cancellation of the context. It looks like this task may require adding an `error` return parameter to the function that `g.AddContext(...)` accepts, too.